### DISCUSSION TITLE
Callback from parser

### DISCUSSION THREAD

**Discussion by rogerl50** [peggyjs/peggy#225] [Q&A] [answered]
I want to parse a string that is generated inside a class instance by calling a peggy generated parser. One of the parsing actions needs to invoke a function defined inside the class instance, so that I have access to the environment of that instance, including `this`. How can I make that happen?

**Comment by Mingun** [comment, marked as answer]
Pass a class instance via [options](https://github.com/peggyjs/peggy#using-the-parser):
```js
let parser = peggy.generate(...);

parser.parse(input, { myClass });
// in the code block in the grammar:
options.myClass.<...>
```
